### PR TITLE
Implemented SignatureData for signatures

### PR DESCRIPTION
**Pull request by jimni1222**
## Proposed changes

This PR introduces `SignatureData` structure to include signature inside.

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [x] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/caver-js/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/caver-js)
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

refer #249 

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: packages/caver-transaction/src/transactionHelper/transactionHelper.js [hunk 1]
```diff
@@ -132,19 +133,23 @@ const refineSignatures = (sigArray, isLegacy = false) => {
     const set = new Set()
     let result = []
 
-    const arrayOfSignatures = !_.isArray(sigArray[0]) ? [sigArray] : sigArray
-
+    let arrayOfSignatures = sigArray
```
Comment (sirano11) [inline packages/caver-transaction/src/transactionHelper/transactionHelper.js:136]: This function needs to modify function description. :)

File: packages/caver-wallet/src/keyring/signatureData.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+    You should have received a copy of the GNU Lesser General Public License
+    along with the caver-js. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+const _ = require('lodash')
+const utils = require('../../../caver-utils')
+
+/**
+ * Representing a PrivateKey class that includes private key string.
```
Comment (sirano11) [inline packages/caver-wallet/src/keyring/signatureData.js:23]:
```suggestion
 * Representing a SignatureData class that includes ECDSA signature data string.
```

File: packages/caver-transaction/src/transactionHelper/transactionHelper.js (hunk 1 above)
Comment (jimni1222) [reply packages/caver-transaction/src/transactionHelper/transactionHelper.js:137]: Thank you :). I modified the description

File: packages/caver-wallet/src/keyring/signatureData.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+const utils = require('../../../caver-utils')
+
+/**
+ * Representing a SignatureData class that includes ECDSA signature data string.
+ * @class
+ */
+class SignatureData {
+    /**
+     * creates a privateKey.
+     * @param {string} key - The private key string.
```
Comment (sirano11) [inline packages/caver-wallet/src/keyring/signatureData.js:29]:
```suggestion
     * @param {Array.<string>|SignatureData} key - The ECDSA signatureData 
```

File: packages/caver-wallet/src/keyring/signatureData.js [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+const _ = require('lodash')
+const utils = require('../../../caver-utils')
+
+/**
+ * Representing a SignatureData class that includes ECDSA signature data string.
+ * @class
+ */
+class SignatureData {
+    /**
+     * creates a privateKey.
```
Comment (sirano11) [inline packages/caver-wallet/src/keyring/signatureData.js:28]:
```suggestion
     * creates a SignatureData.
```

File: packages/caver-wallet/src/keyring/signatureData.js [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+     * @type {string}
+     */
+    get v() {
+        return this._v
+    }
+
+    set v(v) {
+        this._v = utils.makeEven(v)
+    }
+
```
Comment (sirano11) [inline packages/caver-wallet/src/keyring/signatureData.js:55]: /**
     * @type {string}
 */

File: packages/caver-wallet/src/keyring/signatureData.js [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+     * @type {string}
+     */
+    get r() {
+        return this._r
+    }
+
+    set r(r) {
+        this._r = utils.makeEven(r)
+    }
+
```
Comment (sirano11) [inline packages/caver-wallet/src/keyring/signatureData.js:74]: /**
     * @type {string}
     */

File: packages/caver-wallet/src/keyring/signatureData.js [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+    get S() {
+        return this.s
+    }
+
+    set S(s) {
+        this.s = s
+    }
+
+    isEmpty() {
```
Comment (sirano11) [inline packages/caver-wallet/src/keyring/signatureData.js:102]: Don't we need to function description?

File: packages/caver-wallet/src/keyring/signatureData.js [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+        if (this.v === '0x01' && this.r === '0x' && this.s === '0x') return true
+        return false
+    }
+
+    encode() {
```
Comment (sirano11) [inline packages/caver-wallet/src/keyring/signatureData.js:107]: Don't we need to add function description?

File: packages/caver-wallet/src/keyring/signatureData.js [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+        return [this.v, this.r, this.s]
+    }
+
+    toString() {
```
Comment (sirano11) [inline packages/caver-wallet/src/keyring/signatureData.js:111]: Don't we need to add function description?

File: packages/caver-wallet/src/keyring/signatureData.js (hunk 7 above)
Comment (jimni1222) [reply packages/caver-wallet/src/keyring/signatureData.js:116]: I added descriptions!

File: packages/caver-wallet/src/keyring/signatureData.js (hunk 5 above)
Comment (jimni1222) [reply packages/caver-wallet/src/keyring/signatureData.js:55]: I added descriptions!

File: packages/caver-wallet/src/keyring/signatureData.js (hunk 6 above)
Comment (jimni1222) [reply packages/caver-wallet/src/keyring/signatureData.js:77]: I added descriptions!

File: packages/caver-wallet/src/keyring/signatureData.js (hunk 8 above)
Comment (jimni1222) [reply packages/caver-wallet/src/keyring/signatureData.js:126]: I added descriptions!

File: packages/caver-wallet/src/keyring/signatureData.js (hunk 9 above)
Comment (jimni1222) [reply packages/caver-wallet/src/keyring/signatureData.js:135]: I added descriptions!

**Pull request merged**